src/sentry/models/artifactbundle.py

```diff
@@ -175,25 +171,23 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_debugidartifactbundle"
 
-        # We can have the same debug_id pointing to different artifact_bundle(s) because the user might upload
-        # the same artifacts twice, or they might have certain build files that don't change across builds.
-        unique_together = (("debug_id", "artifact_bundle", "source_file_type"),)
```

Comment:
For all the cases where we replace a `unique` constraint with a simple index:

NOTE to other reviewers: We are doing this because table partitioning (which we eventually want to do for quick deletes) is incompatible with unique constraints :-(

However, we should be extra careful with all of these to ensure on the writing side that we are not creating duplicates. As those are not enforced directly on database level, we can never completely remove the likelihood though :-(

Reply:
You might want try use a distributed lock on the write path to help avoid double writes. That'd probably guarantee it - you could just hash the unique attributes together and take out a lock when writing. Take a look around the codebase for `lock.acquire()` for some examples.